src/main/java/com/selfxdsd/selfweb/ReposController.java

```diff
@@ -34,12 +37,28 @@
 @Controller
 public class ReposController {
 
+    /**
+     * Logged user.
+     */
+    private final User user;
+
+    /**
+     * Ctor.
+     * @param user Logged user.
+     */
+    @Autowired
+    public ReposController(final User user) {
+        this.user = user;
+    }
+
     /**
      * Serve the Repos page of Self.
+     * @param model Model.
      * @return Repos page.
      */
     @GetMapping("/repositories")
-    public String index() {
+    public String index(final Model model) {
+        model.addAttribute("provider", user.provider().name());
```

Comment:
@criske I don't think we need this. In the ``<head>`` element we always call ``/users/self`` to check if the user is authenticated or not. In the success callback of that function, we can set a global JS variable. 

The point is, I would like to have as little coupling between front-end and back-end as possible, so we should use Spring MVC only where we have no other choice :D

Reply:
@amihaiemil I need to leave a puzzle for that, there are a couple of places where thymeleaf directives are used; for example in the contributor form
` <label for="username" th:text="${#strings.capitalize(provider) +' Username*'}">Provider Username*</label>`

Just using jQuery would require extra code.

Reply:
@criske right, let's leave a puzzle for this.